src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2298,16 +2298,23 @@ public static T[] FindAll<T>(T[] array, Predicate<T> match)
                 throw new ArgumentNullException(nameof(match));
             }
 
-            LowLevelList<T> list = new LowLevelList<T>();
+            int pos = 0;
+            T[] result = Empty<T>();
             for (int i = 0; i < array.Length; i++)
             {
                 if (match(array[i]))
                 {
-                    list.Add(array[i]);
+                    if (pos == result.Length)
+                        Resize(ref result, pos == 0 ? 4 : pos * 2);
```

Comment:
Would it be worth wrapping this in a `Math.Min`?

```csharp
Resize(ref result, Math.Min(pos == 0 ? 4 : pos * 2, array.Length)); 
```

It would save an allocation if `match()` matches all elements of the array.